### PR TITLE
CI test - supply ZET binary instead of building it every time

### PR DESCRIPTION
**Pull request by qrkourier**
In CI this change uses the binary built by cmake.yml. 

Running locally, you must set a var, e.g., `ZITI_EDGE_TUNNEL_BIN=./build/programs/ziti-edge-tunnel/Release/ziti-edge-tunnel ./docker/docker.test.bash`.

### CODE REVIEW

**Review by dovholuknf — APPROVED** [review]

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -1,13 +1,6 @@
 name: CI build
 
 on:
-  push:
```
Comment (dovholuknf) [inline .github/workflows/cmake.yml:4]: actually, was this change intentional?

Comment (qrkourier) [reply .github/workflows/cmake.yml:4]: Yes. This PR changes the way GitHub workflows are triggered. Before, cmake.yml was triggered by push to default and PR HEADs based on main _and_ as a reusable (callable) workflow on release.

Now, cmake.yml is only called by other workflows, and test-deployments.yml takes over calling cmake.yml on push to default and PR HEADs based on main.

**Comment by qrkourier** [comment]
I expect the checks on this branch to succeed after the branch merges on which this twig is based.

File: .github/workflows/cmake.yml (hunk 1 above)
Comment (dovholuknf) [reply .github/workflows/cmake.yml:4]: Assuming I understand you, this feels inverted to me. i would think the flow is:

- build the artifacts
- upload the artifacts to the job
- kick off other 'test' jobs that download the artifact they want to test/verifiy
- mark the job successful/failed

having something like "test-deployments" initiate the build -- seems wrong. Maybe I'm not understanding you?

Comment (qrkourier) [reply .github/workflows/cmake.yml:4]: Correct. That is the flow. The Test deployments workflow runs on push to default and PR HEADs based on default. We can name it whatever.
